File: lab2.py
import re
_input = open('access.log', 'r')
ip_list = re.findall(r'(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})(?:\.(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})){3}\b', _input.read())
_input.close()
ip_dic = {}
for ip in ip_list:
    if ip in ip_dic:
        ip_dic[ip] += 1
    else:
        ip_dic[ip] = 1
_output = open('output.txt', 'w')
sum = ip_dic.keys()
subnets = set(re.findall(r'[1-2]?[0-9]?[0-9]\.[1-2]?[0-9]?[0-9]\.[1-2]?[0-9]?[0-9]\.', str(sum))) 
for subnet in subnets:
    for subnet_sum in sum:
        if subnet_sum.startswith(subnet):
            _output.write(subnet_sum)
            _output.write(' ')
            _output.write(str(ip_dic[subnet_sum]))
            _output.write('\n')
    _output.write('\n')
_output.close()
print(subnets)

File: lab2.rb
_input = IO.read("access.log")
ip_list = _input.scan(/(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})(?:\.(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})){3}\b/)
ip_dic = Hash.new
for ip in ip_list
	if ip_dic.has_key?(ip) then
		ip_dic[ip] += 1
	else
		ip_dic[ip] = 1
	end
end

fd = IO.sysopen("output.txt", "w")
_output = IO.new(fd, "w")
sum = ip_dic.keys
subnets = sum.join(' ').scan(/[1-2]?[0-9]?[0-9]\.[1-2]?[0-9]?[0-9]\.[1-2]?[0-9]?[0-9]\./).uniq!
for subnet in subnets
	for subnet_sum in sum
		if subnet_sum.start_with?(subnet) then
			_output.print(subnet_sum, ' ', ip_dic[subnet_sum], "\n")
		end
	end
	_output.print("\n")
end